components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,265 @@
+
+    /**
+     * Get the identity property specified in identity-event.properties
+     *
+     * @param moduleName   The name of the module which the property belongs to
+     * @param propertyName The name of the property which should be fetched
+     * @return The required property
+     */
+    public static String getIdentityEventProperty(String moduleName, String propertyName) {
+
+        // Retrieving properties set in identity event properties
+        String propertyValue = null;
+        try {
+            ModuleConfiguration moduleConfiguration = IdentityEventConfigBuilder.getInstance()
+                    .getModuleConfigurations(moduleName);
+
+            if (moduleConfiguration != null) {
+                propertyValue = moduleConfiguration.getModuleProperties().getProperty(propertyName);
+            }
+        } catch (IdentityEventException e) {
+            log.error("An error occurred while retrieving module properties because " + e.getMessage(), e);
+        }
+        return propertyValue;
+    }
+
+    /**
+     * Retrieve event uri.
+     *
+     * @param eventKey  Event key.
+     * @return Event uri string.
+     * @throws IdentityEventServerException If an error occurs.
+     */
+    public static String getEventUri(String eventKey) throws IdentityEventServerException {
+
+        try {
+            ResourceConfig eventConfigObject = EventConfigurationMgtUtils.getEventConfig(eventKey);
+            if (eventConfigObject.getConfigs() != null &&
+                    eventConfigObject.getConfigs().containsKey(EVENT_CONFIG_SCHEMA_NAME_KEY)) {
+                return (String) eventConfigObject.getConfigs().get(EVENT_CONFIG_SCHEMA_NAME_KEY);
+            } else {
+                throw new IdentityEventServerException("Event schema not found in the resource event config " +
+                        "for the eventKey: " + eventKey);
+            }
+        } catch (EventConfigurationMgtServerException e) {
+            throw new IdentityEventServerException(e.getErrorCode(), e.getMessage());
+        } catch (ClassCastException e) {
+            throw new IdentityEventServerException(ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER.getCode(),
+                    ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER.getMessage());
+        }
+    }
+
+    /**
+     * Build the event data provider.
+     *
+     * @param event Event object.
+     * @return Event data object.
+     */
+    public static EventData buildEventDataProvider(Event event) {
+
+        Map<String, Object> properties = event.getEventProperties();
+        Map<String, Object> params = (Map)properties.get("params");
+        AuthenticationContext context = (AuthenticationContext)properties.get("context");
+        Object user = params.get("user");
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        // User information in login failed events.
+        User loginIdentifier = new User();
```

Comment:
Why do we create this user instance ?

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/6e511b91f52c4b71ccff73c989894c76b285ebf0